src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -418,9 +434,17 @@ public void Reverse()
         public void Reverse(int index, int count)
         {
             if (index < 0)
+            {
                 throw new ArgumentOutOfRangeException(nameof(index));
+            }
             if (count < 0)
+            {
                 throw new ArgumentOutOfRangeException(nameof(count));
+            }
+            if (Count - index < count)
+            {
+                throw new ArgumentException(); // Argument_InvalidOffLen
```

Comment:
SR does not contain such or similar message. Also I tried add string to Strings.rex file but Visual Studio does not open it and return some moniker error.
I will create issue about this after merge and somebody who knows how to add resource strings will make such changes later.